nomos-da/network/core/src/protocols/replication/behaviour.rs

```diff
@@ -289,10 +310,10 @@ where
     /// sent again.
     fn send_message_impl(&mut self, waker: Option<&Waker>, message: &DaMessage) {
         let message_id = (message.share.blob_id.to_vec(), message.subnetwork_id);
```

Comment:
I think `ReplicationRequest` could have `id()` method that returns `ReplicationMessageId` type of `[u8;34]` that the cache would expect as it's key.